fadin/solver.py

```diff
@@ -148,72 +146,56 @@ def __init__(self, n_dim, kernel, init='random', optim_mask=None,
-        random_bl_init = init == 'random' or self.moment_matching
-        if random_bl_init or init['baseline'] is None:
-            self.baseline = torch.rand(self.n_dim)
-        else:
-            self.baseline = init['baseline'].float()
         if optim_mask['baseline'] is None:
             self.baseline_mask = torch.ones([n_dim])
         else:
-            assert optim_mask['baseline'].shape == self.baseline.shape, \
+            assert optim_mask['baseline'].shape == torch.Size([n_dim]), \
                 "Invalid baseline_mask shape, must be (n_dim,)"
             self.baseline_mask = optim_mask['baseline']
-        bl = self.baseline * self.baseline_mask
-        self.baseline = bl.requires_grad_(True)
-
-        if init == 'random' or self.moment_matching or init['alpha'] is None:
-            self.alpha = torch.rand(self.n_dim, self.n_dim)
-        else:
-            self.alpha = init['alpha'].float()
         if optim_mask['alpha'] is None:
             self.alpha_mask = torch.ones([self.n_dim, self.n_dim])
         else:
-            assert optim_mask['alpha'].shape == self.alpha.shape, \
+            assert optim_mask['alpha'].shape == torch.Size([n_dim, n_dim]), \
                 "Invalid alpha_mask shape, must be (n_dim, n_dim)"
             self.alpha_mask = optim_mask['alpha']
-        self.alpha = (self.alpha * self.alpha_mask).requires_grad_(True)
-
-        if init == 'random' or self.moment_matching or init['kernel'] is None:
-            kernel_params_init = init_kernel_parameters(kernel,
-                                                        self.kernel_length,
-                                                        self.n_dim)
-
-        self.kernel_params_fixed = kernel_params_init
 
-        self.n_kernel_params = len(kernel_params_init)
-
-        self.kernel_model = DiscreteKernelFiniteSupport(self.delta, self.n_dim,
-                                                        kernel, self.kernel_length,
-                                                        0, self.kernel_length,
-                                                        grad_kernel)
-        self.kernel = kernel
-        # Set optimizer
-        self.params_intens = [self.baseline, self.alpha]
-
-        for i in range(self.n_kernel_params):
-            self.params_intens.append(
-                kernel_params_init[i].float().clip(1e-4).requires_grad_(True))
+        # Model parameters
+        if 'moment_matching' in init:
+            self.init_mode = 'moment_matching'
+            self.mm_mode = init.split('_')[-1]
+        elif init == 'random':
+            self.init_mode = 'random'
+        else:
+            self.init_mode = init
+            assert 'baseline' in init.keys(), \
+                "baseline parameter must be given"
+            assert 'alpha' in init.keys(), \
+                "alpha parameter must be given"
+            assert 'kernel' in init.keys(), \
+                "kernel parameter must be given"
```

Comment:
This  should be mostly done in `inti_hawkes_params`

here, you should only have param validation:
```python
if isinstance(init, str):
    assert init in [....]
else:
     assert isinstance(init, dict) and have the right key.
```

Reply:
done in [commit 2f5b2a0](https://github.com/mind-inria/FaDIn/pull/16/commits/2f5b2a05ccf45dab43c6d0c639ddff7aa3865758) :)